docs/src/references/mqtt-api.md

```diff
@@ -0,0 +1,836 @@
+
+|Segment|Description|
+|-------|-----------|
+|&lt;category&gt;|Category of data (telemetry or commands) which is related to the **identifier**. A fixed list of categories is used to represent data types such as; measurements, alarms, events, commands etc.|
+|&lt;type&gt;|A unique type/name used to identify the information being pushed or received. Types allow users to filter/subscribe to data which interests them. For example, a measurement could be published under a type called "flow_rate", then other clients interested in the flow rate can subscribe to that single typed topic.|
+|`...`|Additional channel specific topic segments. Each **category** is responsible for defining the number and meaning of the remaining topic segments.|
+
+
+#### Categories
+
+The following is an overview of the channel categories which are available.
+
+|Category|Description|
+|--------|-------|
+|m|Measurements|
+|e|Events|
+|a|Alarms|
+|cmd|Commands|
+
+
+## Entity registration
+
+Since thin-edge doesn't enforce what each entity identification level means,
+an explicit registration is required to register every entity that is going to send data or receive commands.
+For example, before a measurement can be sent from a service named `tedge-agent` from the device `rpi1001`,
+the entity named `rpi1001` must be registered as a `device`,
+and `tedge-agent` must be registered as a `service` linked to that device.
+
+An entity can be registered with thin-edge by publishing a retained message to the entity identification topic prefix
+with the entity type and other metadata that defines that entity.
+To model the example mentioned above, if an entity identification topic scheme like the following is used:
+
+```text
+te/device/<device_id>/service/<service_id>
+```
+
+### Examples
+
+Here are a few examples of how various entities can be registered.
+
+#### Register a device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main' '{
+  "@type": "device",
+  "type": "Gateway"
+}'
+```
+
+Or use the a device can be registered using an explicit id:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main' '{
+  "@type": "device",
+  "@id": "tedge001",
+  "type": "Gateway"
+}'
+```
+
+#### Register a service of the main device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/service/nodered' '{
+  "@type": "service",
+  "displayName": "nodered",
+  "type": "systemd"
+}'
+```
+
+The service is implicitly linked to the parent derived from the topic, `main` in this example.
+
+But the parent can be explicitly provided as well with the `@parent` key,
+if the parent can not be derived from the topic directly:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/component_namespace/service/nodered/instance-1' '{
+  "@type": "service",
+  "@parent": "te/device/main",
+  "displayName": "nodered",
+  "type": "systemd"
+}'
+```
+
+#### Register a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01' '{
+  "@type": "child-device",
```

Comment:
It is assumed that without the `@parent` property, the child device is a child of the main device. It is more of a convenience though instead of being explicit:

```
tedge mqtt pub -r 'te/device/child01' '{
  "@type": "child-device",
  "@parent": "te/device/main",
  "displayName": "child01",
  "type": "SmartHomeHub"
}'
```

Reply:
I guess there is a little influence from Cumulocity here...because registering a child device, implicitly registers the child device under the digital twin which is linked to the device certificate common name....So you never have to declare that you are a child of the main device, you automatically are.

The main device has a bit of a special meaning, so technically it exists without having to register it locally as it is created as part of the `tedge connect c8y` command. Though I would be surprised if the other cloud implementations did not have a similar concept given that the device certificate plays a pivotal role in identifying the device on the cloud side.